### PR TITLE
remove StdErrLog from JettyHttpServer

### LINKED ISSUE #8108: [3.0] Rest协议下错误日志丢失
- [x] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.
- [x] I have checked the [FAQ](https://github.com/apache/dubbo/blob/master/FAQ.md) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 3.0.0
* Operating System version: Mac OS Big Sur 11.4
* Java version: 1.8

### Steps to reproduce this issue

1. 克隆dubbo代码，检出3.0版本的tag；
2. 修改demo工程，制造1个rest协议报错的case；
3. 启动demo工程 /dubbo-demo/dubbo-demo-xml/dubbo-demo-xml-provider；
4. 观察控制台日志，发现有部分错误日志被打印到了StdError，截图如下：

image

### Expected Result

将错误日志桥接到Dubbo自己的日志系统，而非打印到StdError。

### Actual Result

错误日志打印到了StdError，导致基于Dubbo日志做的收集和其它扩展失效。

关键代码：

1. 类文件： ``org.apache.dubbo.remoting.http.jetty.JettyHttpServer``；
2. 代码位置：第33行 ``Log.setLog(new StdErrLog());``。

### PR DESCRIPTION
**Pull request by huangwenkan9**
## What is the purpose of the change
fix #8108

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by wangchengming666 — CHANGES_REQUESTED** [review]
pls add license

File: dubbo-remoting/dubbo-remoting-http/src/main/java/org/apache/dubbo/remoting/http/jetty/JettyLoggerAdapter.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+    private String format(String msg, Object... args) {
+        msg = String.valueOf(msg);
+        String braces = "{}";
+        StringBuilder builder = new StringBuilder();
+        int start = 0;
+        Object[] var6 = args;
+        int var7 = args.length;
+
+        for(int var8 = 0; var8 < var7; ++var8) {
```
Comment (CrazyHZM) [inline dubbo-remoting/dubbo-remoting-http/src/main/java/org/apache/dubbo/remoting/http/jetty/JettyLoggerAdapter.java:141]: Enhance the readability of variables.

File: dubbo-remoting/dubbo-remoting-http/src/main/java/org/apache/dubbo/remoting/http/jetty/JettyLoggerAdapter.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    }
+
+    @Override
+    public void ignore(Throwable throwable) {
+        if (logger.isDebugEnabled()){
+            logger.debug("IGNORED EXCEPTION ", throwable);
+        }
+    }
+
+    private String format(String msg, Object... args) {
```
Comment (CrazyHZM) [inline dubbo-remoting/dubbo-remoting-http/src/main/java/org/apache/dubbo/remoting/http/jetty/JettyLoggerAdapter.java:133]: Provide unit test

File: dubbo-remoting/dubbo-remoting-http/src/main/java/org/apache/dubbo/remoting/http/jetty/JettyLoggerAdapter.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+package org.apache.dubbo.remoting.http.jetty;
+
+import org.apache.dubbo.common.logger.LoggerFactory;
+import org.eclipse.jetty.util.log.AbstractLogger;
+import org.eclipse.jetty.util.log.Logger;
+
+/**
+ * logger adapter for jetty
+ */
+public class JettyLoggerAdapter extends AbstractLogger {
```
Comment (CrazyHZM) [inline dubbo-remoting/dubbo-remoting-http/src/main/java/org/apache/dubbo/remoting/http/jetty/JettyLoggerAdapter.java:26]: Provide unit test

File: dubbo-remoting/dubbo-remoting-http/src/main/java/org/apache/dubbo/remoting/http/jetty/JettyLoggerAdapter.java (hunk 2 above)
Comment (huangwenkan9) [reply dubbo-remoting/dubbo-remoting-http/src/main/java/org/apache/dubbo/remoting/http/jetty/JettyLoggerAdapter.java:133]: how to test private method using mockito

File: dubbo-remoting/dubbo-remoting-http/src/test/resources/log4j.xml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<!DOCTYPE log4j:configuration SYSTEM "log4j.dtd">
+<log4j:configuration xmlns:log4j="http://jakarta.apache.org/log4j/">
+    <!-- ===================================================================== -->
+    <!-- 以下是appender的定义 -->
+    <!-- ===================================================================== -->
```
Comment (CrazyHZM) [inline dubbo-remoting/dubbo-remoting-http/src/test/resources/log4j.xml:22]: please change to English description

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**